libs/wingsdk/src/shared-aws/function.inflight.ts

```diff
@@ -11,12 +13,49 @@ export class FunctionClient implements IFunctionClient {
   /**
    * Invoke the function, passing the given payload as an argument.
    */
-  public async invoke(payload: string): Promise<string> {
+  public async invoke(
+    payload: string,
+    logsCollector?: Trace[]
+  ): Promise<string> {
     const command = new InvokeCommand({
       FunctionName: this.functionArn,
       Payload: fromUtf8(JSON.stringify(payload)),
+      LogType: LogType.Tail,
     });
     const response = await this.lambdaClient.send(command);
+    const arnParts = this.functionArn.split(":");
+    const functionName = arnParts[arnParts.indexOf("function") + 1];
+
+    if (logsCollector) {
+      const logs: string[] =
+        atob(response.LogResult ?? "").match(
+          // date, then any char then a line of winglog
+          /(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z)(.*)winglogstart:[\s\S]+?:winglogend/gm
+        ) ?? [];
+
+      logsCollector.push(
+        ...logs.map((item: string) => ({
+          data: {
+            message: (item.match(/(?<=winglogstart:)[\s\S]*?(?=:winglogend)/) ??
+              [])[0],
+          },
```

Comment:
I agree, but I think we should let the user choose, (if they got a test calling to 100 cloud function, that only 20 of them log something, instead of getting 20 lines, it will get messy pretty soon), I think I'll add a flag, what about `-log=full | exact | none` with the default to `exact`? (I really have no idea what it should be called)

Reply:
update: removed